documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+
+
+====== directory-based index manager [[query.directory-based]]
+
+This is the default Index Manager used when no index manager is configured. The `directory-based` index manager is used to manage indexes backed by a local lucene directory. It supports _ram_, _filesystem_ and non-clustered
+_infinispan_ storage.
+
+
+[small]*Filesystem storage*
+
+This is the default storage, and used when index manager configuration is omitted. The index is stored in the filesystem using a link:https://lucene.apache.org/core/6_0_0/core/org/apache/lucene/store/MMapDirectory.html[MMapDirectory].
+It is the recommended storage for local indexes. Although indexes are persistent on disk, they get memory mapped by Lucene and thus offers decent query performance.
+
+Configuration:
+
+[source,xml]
+----
+<replicated-cache name="myCache">
+   <indexing index="ALL">
+      <!-- Optional: define base folder for indexes -->
+      <property name="default.indexBase">${java.io.tmpdir}/baseDir</property>
+   </indexing>
+</replicated-cache>
+----
+
+Infinispan will create a different folder under `default.indexBase` for each entity (index) present in the cache.
+
+[small]*Ram storage*
+
+Index is stored in memory using a link:https://lucene.apache.org/core/6_0_0/core/org/apache/lucene/store/RAMDirectory.html[Lucene RAMDirectory].
+Not recommended for large indexes or highly concurrent situations. Indexes stored in Ram are not persistent, so after a cluster shutdown a <<query.massindexer,re-index>>
+is needed. Configuration:
+
+
+[source,xml]
+----
+<replicated-cache name="myCache">
+   <indexing index="ALL">
+      <property name="default.directory_provider">ram</property>
+   </indexing>
+</replicated-cache>
+----
+
+[small]*Infinispan storage*
+
+In order to use Infinispan storage with a non-shared index, it's necessary to use LOCAL caches for the indexes:
+
+[source,xml]
+----
+<replicated-cache name="default">
+    <indexing index="ALL">
+        <property name="default.locking_cachename">LuceneIndexesLocking_custom</property>
+        <property name="default.data_cachename">LuceneIndexesData_custom</property>
+        <property name="default.metadata_cachename">LuceneIndexesMetadata_custom</property>
+    </indexing>
+</replicated-cache>
+
+<local-cache name="LuceneIndexesLocking_custom">
+    <indexing index="NONE" />
+</local-cache>
+
+<local-cache name="LuceneIndexesMetadata_custom">
+    <indexing index="NONE" />
+</local-cache>
+
+<local-cache name="LuceneIndexesData_custom">
+    <indexing index="NONE" />
+</local-cache>
+
+----
+
+
+====== near-real-time index manager
+
+Similar to the `directory-based` index manager but takes advantage of the Near-Real-Time features of Lucene. It has better write performance than
+the `directory-based` because it flushes the index to the underlying store less often. The drawback is that it can cause loss of data in case
```

Comment:
s/data/ index updates/ ? Just to clarify that we wouldn't lose *Infinispan data*.

I think the NRT is usually a good option so I'd be careful to not scare people off.